docs/exporting-config.md

```diff
@@ -0,0 +1,61 @@
+# Exporting and Importing Configuration
+
+Drush provides commands to export and import the configuration files
+to and from a Drupal 8 site.  Configuration can be altered by different
+methods in order to provide different behaviors in different environments;
+for example, a development server might be configured slightly differently
+than the production server.
+
+This document describes how to make simple value changes to configuration
+based on the environment, how to have a different set of enabled modules
+in different configurations without affecting your exported configuration
+values, and how to make more complex changes.
+
+## Simple Value Changes
+
+It is not necessary to alter the configuration system values to 
+make simple value changes to configuration variables, as this may be
+done by the configuration override system.
+
```

Comment:
Lets link to https://github.com/gherkins/regexpbuilderphp. Thats a really good doc page. We might even summarize its contents here (I'm thinking that module provided overrides can be relevant).